### PR TITLE
Added box and download icon for static files; modified the current md…

### LINKED ISSUE #1065: [website] Add download icon next to file names
**What would you like to be added**:

We have a shortcode that writes the contents of a static file into the website, along with a link for downloading it.
There should be:
- an icon before the file name.
- a box around the name

https://kubernetes.io/docs/concepts/workloads/controllers/job/#running-an-example-job

Perhaps we can adapt the shortcode that kubernetes uses https://github.com/kubernetes/website/blob/3c9c3afd2abd086f79b7d220884f39747d06496c/layouts/shortcodes/code.html

**Why is this needed**:

An icon would make it clear that it's a link for downloading the file

**Completion requirements**:

This enhancement requires the following artifacts:

- [ ] Design doc
- [ ] API change
- [ ] Docs update

The artifacts should be linked in subsequent comments.

### PR DESCRIPTION
**Pull request by FZhg**
… file to use the new include shortcode

#### What type of PR is this?

/kind documentation

#### Which issue(s) this PR fixes:

Fixes #1065 

#### Special notes for your reviewer:
- [x]  Design doc
image

- [x] API change

##### Include static code file
To embed an static code file into the a content page of the website, just add the the following code inside the page html file.
```
{{< include file="examples/<your-filepath>" lang="<your-code-language>" >}}
```
or without named parameters
```
{{< include "examples/<your-filepath>" "<your-code-language>" >}}
```

##### Include non-code static file
If you want include a non-code file,
```
{{<  include "<your-filepath>" >}}
```
##### Developer Notes
The first parameter of the `include` shortcode is a relative path to the `static` directory.  Note that you shouldn't use `{{% %}}` to call this shortcode, which could result in ill-formatted codeblocks. You should only use `{{<  >}}` to call this shortcode. 

- [x] Docs update
 I updated the relative path parameter of all usages of include in this page: 
    - [x] install-kueue-queues.py
    - [x] sample-job.py
    - [x] sample-queue-control.py
    - [x] sample-flux-operator-job.py
    - [x] sample-mpijob.py
    - [x] sample-pytorchjob.yaml
    - [x] sample-tfjob.yaml 
 
The reviewers can check [run python jobs](https://deploy-preview-1118--kubernetes-sigs-kueue.netlify.app/docs/tasks/run_python_jobs/), [run tf jobs](https://deploy-preview-1118--kubernetes-sigs-kueue.netlify.app/docs/tasks/run_kubeflow_jobs/run_tfjobs/) and [run a PyTorch Job](https://deploy-preview-1118--kubernetes-sigs-kueue.netlify.app/docs/tasks/run_kubeflow_jobs/run_pytorchjobs/).

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by tenzen-y** [comment]
@FZhg Thank you for creating this PR.
Why did you open this PR separately with https://github.com/kubernetes-sigs/kueue/pull/1106?

**Comment by FZhg** [comment]
> @FZhg Thank you for creating this PR.
> 
> Why did you open this PR separately with https://github.com/kubernetes-sigs/kueue/pull/1106?
> 
> 
Hi! 
I encountered a weird bug in that branch, even after reset. [slack thread](https://kubernetes.slack.com/archives/C1J0BPD2M/p1694577732804749)

So I start another branch from scratch.

**Comment by tenzen-y** [comment]
> > @FZhg Thank you for creating this PR.
> > Why did you open this PR separately with #1106?
> 
> Hi! Thanks for tagging my PR! I encountered a weird bug in that branch, even after reset. [slack thread](https://kubernetes.slack.com/archives/C1J0BPD2M/p1694577732804749)
> 
> So I start another branch from scratch.

Ah, I see. Thanks for the clarifications.

**Comment by tenzen-y** [comment]
/test pull-kueue-test-integration-main

Due to #1122

**Comment by FZhg** [comment]
> As I can see, the new form doesn't apply all examples. Can you apply new form to all examples?

Thanks for pointing this out!  Could you tell me which specific examples that I missed?
I think I did replace all the usage of the included shortcode though.

**Comment by FZhg** [comment]
> As I can see, the new form doesn't apply all examples. Can you apply new form to all examples?

Do you mean the duplicate code examples? This is fixed by #1078.
The changes in #1078 are compatible with this PR.

**Comment by tenzen-y** [comment]
> > As I can see, the new form doesn't apply all examples. Can you apply new form to all examples?
> 
> Do you mean the duplicate code examples? This is fixed by #1078. The changes in #1078 are compatible with this PR, except that we need to substitute `{{% %}}` with `{{< >}}`. Could you merge #1078 first, then I will fetch and update them all.

Oh, I see. Yes, I am confused with #1078. I think we can merge this PR into the main without being blocked by #1078.
Let me check this patch.

**Comment by FZhg** [comment]
> > > As I can see, the new form doesn't apply all examples. Can you apply new form to all examples?
> > 
> > 
> > Do you mean the duplicate code examples? This is fixed by #1078. The changes in #1078 are compatible with this PR, except that we need to substitute `{{% %}}` with `{{< >}}`. Could you merge #1078 first, then I will fetch and update them all.
> 
> Oh, I see. Yes, I am confused with #1078. I think we can merge this PR into the main without being blocked by #1078. Let me check this patch.

That would be great! Thanks for reviewing!

**Comment by alculquicondor** [comment]
Could you center the content vertically and add some padding on the right side?

**Comment by FZhg** [comment]
> Could you center the content vertically and add some padding on the right side?

Thanks for raising this up! Could you tell me more about this? Do you mean that I need to center the download icon and filename vertically or the code blocks?  with the right padding, it should look like this:
image

Could you confirm this? Thanks!

**Comment by alculquicondor** [comment]
> > Could you center the content vertically and add some padding on the right side?
> 
> Thanks for raising this up! Could you tell me more about this? Do you mean that I need to center the download icon and filename vertically or the code blocks? with the right padding, it should look like this: image
> 
> Could you confirm this? Thanks!
> 

Yes, this looks good to me.

> P.s. The k8s website static file style looks like this: image

That looks like it's missing some style. You can consider contributing to github.com/kubernetes/website after this :) But that's outside of my ownership.

**Comment by FZhg** [comment]
> > > Could you center the content vertically and add some padding on the right side?
> > 
> > 
> > Thanks for raising this up! Could you tell me more about this? Do you mean that I need to center the download icon and filename vertically or the code blocks? with the right padding, it should look like this: image
> > Could you confirm this? Thanks!
> 
> Yes, this looks good to me.
> 
> > P.s. The k8s website static file style looks like this: image
> 
> That looks like it's missing some style. You can consider contributing to github.com/kubernetes/website after this :) But that's outside of my ownership.

I did the style change. Here is the style now:
image

Could you run the test again for generating the preview link? Thanks!

**Comment by tenzen-y** [comment]
Thank you! It looks good to me.
/approve

leaving lgtm on @alculquicondor

**Comment by FZhg** [comment]
@alculquicondor Hi, Commits are stashed.

**Comment by alculquicondor** [comment]
That looks great!

Since #1078, could you do another pass to ensure all samples are using the new shortcode?

**Comment by FZhg** [comment]
> That looks great!
> 
> Since #1078 already merged, could you do another pass to ensure all samples are using the new shortcode?


Thanks for your comments! I have merged the main branch and updated all usages. I am really excited to have my first PR merged at k8s! You can check the preview links here:

- [x]  [run jobs](https://deploy-preview-1118--kubernetes-sigs-kueue.netlify.app/docs/tasks/run_jobs/#1-define-the-job)
- [x] [run mpi jobs](https://deploy-preview-1118--kubernetes-sigs-kueue.netlify.app/docs/tasks/run_kubeflow_jobs/run_mpijobs/#sample-mpi-job)
- [x] [run xgb boost jobs](https://deploy-preview-1118--kubernetes-sigs-kueue.netlify.app/docs/tasks/run_kubeflow_jobs/run_xgboostjobs/#sample-xgboostjob)
- [x] [run ray jobs](https://deploy-preview-1118--kubernetes-sigs-kueue.netlify.app/docs/tasks/run_rayjobs/#example-rayjob)

File: site/content/en/docs/tasks/run_kubeflow_jobs/run_mpijobs.md [hunk 1]
```diff
@@ -52,6 +52,6 @@ By default, Kueue will set `suspend` to true via webhook and unsuspend it when t
 
 This example is based on https://github.com/kubeflow/mpi-operator/blob/ccf2756f749336d652fa6b10a732e241a40c7aa6/examples/v2beta1/pi/pi.yaml.
 
-{{% include "jobs/sample-mpijob.yaml" "yaml" %}}
+{{% include "examples/jobs/sample-mpijob.yaml" "yaml" %}}
```
Comment (tenzen-y) [inline site/content/en/docs/tasks/run_kubeflow_jobs/run_mpijobs.md:55]:
```suggestion
{{< include "examples/jobs/sample-mpijob.yaml" "yaml" >}}
```

It this correct?

File: site/content/en/docs/tasks/run_python_jobs.md [hunk 2]
```diff
@@ -326,7 +325,7 @@ Use:
 "kubectl get jobs" to see jobs
 ```
 
-{{% include "python/sample-mpijob.py" "python" %}}
+{{< include "examples/python/sample-mpijob.py" "python" >}}
```
Comment (tenzen-y) [inline site/content/en/docs/tasks/run_python_jobs.md:328]: This is just a confirmation. Why do we need to use a different format? 
`{{< include file="examples/<your-filepath>" lang="<your-code-language>" >}}` vs `{{< include "examples/<your-filepath>" "<your-code-language>" >}}`

Can we use the same format across the whole of the document to improve maintainability?

**Review by tenzen-y — COMMENTED** [review]
@FZhg Thank you for the updates! I really appreciate your effort :)
I left a few comments for confirmation.

File: site/content/en/docs/tasks/run_python_jobs.md (hunk 2 above)
Comment (FZhg) [reply site/content/en/docs/tasks/run_python_jobs.md:328]: I thought about this for a bit. I don't have a good reason to use two different formats. I just used two different formats in the md files because my shortcode implementation supports. For the goal of being maintainable, I am very happy to only keep one of them inside the documents. 

`{{< include file="examples/<your-filepath>" lang="<your-code-language>" >}}` used named parameter. It is verbose but clear on meaning of the parameters. Also the order of the parameters doesn't matter anymore. 

The second one, `{{< include "examples/<your-filepath>" "<your-code-language>" >}}`, is more concise. However, we have to pass the parameters in the correct order. 

which one would you want to keep?

Comment (alculquicondor) [reply site/content/en/docs/tasks/run_python_jobs.md:328]: I like the verbosity, but let's be consistent.

Comment (alculquicondor) [reply site/content/en/docs/tasks/run_python_jobs.md:328]: Let's merge as is for now

/lgtm
/approve
/hold cancel
